botorch/models/gpytorch.py

```diff
@@ -446,17 +447,20 @@ def posterior(
             mvn = self(X)
             mvn = self._apply_noise(X=X, mvn=mvn, observation_noise=observation_noise)
             if self._num_outputs > 1:
-                mean_x = mvn.mean
-                covar_x = mvn.lazy_covariance_matrix
-                output_indices = output_indices or range(self._num_outputs)
-                mvns = [
-                    MultivariateNormal(
-                        mean_x.select(dim=output_dim_idx, index=t),
-                        covar_x[(slice(None),) * output_dim_idx + (t,)],
-                    )
-                    for t in output_indices
-                ]
-                mvn = MultitaskMultivariateNormal.from_independent_mvns(mvns=mvns)
+                if _get_tracing_state():
+                    mvn = MultitaskMultivariateNormal.from_batch_mvn(mvn, task_dim=0)
```

Comment:
Is `task_dim=0` always correct here? Output dim is typically added as the right-most batch dim of the model. Should it be `output_dim_idx`?

Reply:
Yes, I think you're right. the `task_dim` need not be 0 in every case. A more appropriate setting is `output_dim_idx`. This has been fixed now.